appservice/src/TunnelProxy.ts

```diff
@@ -161,20 +184,31 @@ export class TunnelProxy {
 
             const statusCallback: request.RequestCallback = (error: string, response: request.Response, body: string): void => {
                 if (error) {
-                    ext.outputChannel.appendLine(`[WebApp Tunnel] Checking status, error: ${error}`);
-                    reject();
+                    reject({ shouldRetry: false, message: `Error getting tunnel status: ${error}` });
                 } else if (response.statusCode === 200) {
                     ext.outputChannel.appendLine(`[WebApp Tunnel] Checking status, body: ${body}`);
 
-                    // SUCCESS:2222 means that the default ssh tunnel has not been replaced yet
-                    if (isString(body) && body.startsWith('SUCCESS') && body !== 'SUCCESS:2222') {
-                        resolve();
+                    // tslint:disable-next-line:no-unsafe-any
+                    const tunnelStatus: ITunnelStatus = JSON.parse(body);
```

Comment:
What happens if the body fails to parse for some reason? Does the promise ever reject or resolve?

Can we convert this to use async/await with "request-promise" instead of a callback? It's already used at least once in this repo here: https://github.com/Microsoft/vscode-azuretools/blob/master/appservice/src/functionsAdminRequest.ts#L16

Reply:
Sure, I've switched to "request-promise" and it's a lot cleaner now :smile: